### PR TITLE
test: add e2e tests for Input and Input-group 

### PR DESCRIPTION
**Pull request by AntonOlkhovskyi**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/3732
#### Please provide a brief summary of this pull request.
Add e2e test input and input-group components
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: e2e/wdio/platform/tests/input-group.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        webDriver.waitForDisplayed(inputGroupPage.withFormInput);
+        webDriver.scrollIntoView(inputGroupPage.withFormInput);
+        webDriver.mouseHoverElement(inputGroupPage.withFormInputQuestionMark);
+
+        expect(webDriver.getText(inputGroupPage.withFormInputInfoTooltip)).toBe('This is tooltip to help');
+        // return;
+        // }
+    });
+
+
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/input-group.e2e-spec.ts:141]: formatting

File: e2e/wdio/platform/pages/input-group.po.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    withFormInputLabel = 'fdp-input-group-form-example label > span';
+    withFormInputQuestionMark = 'fdp-input-group-form-example label span[role="alert"]';
+    withFormInputAsterixMark = 'fdp-input-group-form-example label';
+    withFormInputErrorTooltip = '[type="error"] span';
+    withFormInputInfoTooltip = '.fd-inline-help__content';
+
+    exampleAreaContainersArr = '.fd-doc-component';
+    rtlSwitcherArr = 'rtl-switch .fd-switch__handle';
+
+
```
Comment (Seamoo13) [inline e2e/wdio/platform/pages/input-group.po.ts:46]: extra empty lines

File: e2e/wdio/platform/pages/input.po.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    requiredInputLabel = 'fdp-platform-input-reactive-form-validation-example .fd-form-label--required';
+    questionMarkSpan = 'span[role="alert"]';
+    inputsLabels = '.fd-section label span.ng-star-inserted';
+    inputsArray = 'input.fd-input';
+    autocompleteInput = 'input#form-input-7';
+    autocompleteInputLabel = 'fdp-platform-input-auto-complete-form-validation-example label';
+    autocompleteOptions = '.fd-popover__popper li';
+
+
+
```
Comment (Seamoo13) [inline e2e/wdio/platform/pages/input.po.ts:28]: extra empty lines

File: e2e/wdio/platform/tests/input-group.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
+import { webDriver } from '../../driver/wdio';
+import { InputGroupPo } from '../pages/input-group.po';
+import inputGroupPContent from '../fixtures/appData/input-group-page-contents';
+import inputGroupTestData from '../fixtures/testData/input-group';
+
+describe('Input Group should', function() {
+
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/input-group.e2e-spec.ts:7]: maybe it's better not to have the extra line before const?

File: e2e/wdio/platform/tests/input-group.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    beforeAll(() => {
+        inputGroupPage.open();
+    });
+
+    afterEach(() => {
+        webDriver.refreshPage();
+    });
+
+    it('have associated label element to describe its purpose', () => {
+        expect(webDriver.getTextArr(inputGroupPage.standartInputLabelsArr)).toEqual(inputGroupPContent.standardInputLabels);
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/input-group.e2e-spec.ts:19]: extra space/line?

File: e2e/wdio/platform/fixtures/appData/input-group-page-contents.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    standardInputPlaceholders: [
+        'Enter value',
+        'Enter value',
+        'Enter value',
+        'Enter value',
+        'Email'
+    ],
+    inputWithFormLabel: 'Quantity',
+    inputWithFormPlaceholder: 'Things quantity',
+
```
Comment (Seamoo13) [inline e2e/wdio/platform/fixtures/appData/input-group-page-contents.ts:18]: not needed empty line

File: e2e/wdio/platform/fixtures/testData/input-group.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export default {
+    numeric_value: '1234',
+    string_value: 'test',
+    email_value: 'test@test.com'
+
```
Comment (Seamoo13) [inline e2e/wdio/platform/fixtures/testData/input-group.ts:5]: not needed empty line

**Review by Seamoo13 — APPROVED** [review]

File: e2e/wdio/platform/tests/input-group.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+        webDriver.setValue(inputGroupPage.compactGroupInput, inputGroupTestData.string_value);
+
+        expect(webDriver.getText(inputGroupPage.compactGroupLeftTextAddon)).toContain('$');
+        expect(webDriver.getText(inputGroupPage.compactGroupButtonAddon)).toContain('Submit');
+        expect(webDriver.isElementClickable(inputGroupPage.compactGroupButtonAddon)).toBe(true);
+        expect(webDriver.getValue(inputGroupPage.compactGroupInput)).toBe(inputGroupTestData.string_value);
+    });
+
+    it('compact be smaller than the default', () => {
+        const defaultHeight = webDriver.getElementSize(inputGroupPage.leftAlignedTextInput) as WebdriverIO.SizeReturn;
```
Comment (valorkin) [inline e2e/wdio/platform/tests/input-group.e2e-spec.ts:83]: shouldn't be `WebdriverIO.SizeReturn` a return type of `getElementSize`

File: e2e/wdio/platform/tests/input-group.e2e-spec.ts [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+        // TODO: Uncomment after merge
+        /*        if (webDriver.isBrowser('Safari')) {
+            expect(webDriver.getText(inputGroupPage.withFormInputErrorTooltip)).toEqual('Value is required');
+            return;
+        }
+        console.log('Skip hover check for Safari');*/
+    });
+
+    it('with form input have info tooltip', () => {
+        if (!webDriver.isIEorSafari()) {
```
Comment (valorkin) [inline e2e/wdio/platform/tests/input-group.e2e-spec.ts:130]: use fast exits to reduce nesting

**Review by valorkin — CHANGES_REQUESTED** [review]
and remove webDriver object as discussed please

File: e2e/wdio/driver/wdio.ts [hunk 10]
```diff
@@ -1,226 +1,271 @@
...
+export function isBrowser(browserName: string): boolean {
+    return browser.capabilities.browserName === browserName;
+}
 
-    isBrowser(browserName: string): boolean {
-      return browser.capabilities.browserName === browserName;
+export function browserIsIEorSafari(): boolean {
+    if (browserIsSafari()) {
+        return true;
+    } else if (browserIsIE) {
```
Comment (Seamoo13) [inline e2e/wdio/driver/wdio.ts:24]: this function doesn't work. please recheck it. 
fixed in my new branch with 
if ( this.isBrowser('Safari') { 
return true
} else if (this.isBrowser('internet explorer) {
return true
}
return false

**Review by Seamoo13 — CHANGES_REQUESTED** [review]

File: e2e/wdio/driver/wdio.ts (hunk 10 above)
Comment (Seamoo13) [reply e2e/wdio/driver/wdio.ts:24]: probably could do it with if ( browserIsSafari() === true) ...

File: e2e/wdio/platform/tests/input-group.e2e-spec.ts (hunk 8 above)
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/input-group.e2e-spec.ts]: Done.

File: e2e/wdio/core/pages/core-base-component.po.ts [hunk 11]
```diff
@@ -1,11 +1,8 @@
-import { webDriver } from '../../driver/wdio';
-
 export class CoreBaseComponentPo {
```
Comment (Seamoo13) [inline e2e/wdio/core/pages/core-base-component.po.ts:1]: need to import open from '../../driver/wdio

**Review by Seamoo13 — APPROVED** [review]

**Review by valorkin — APPROVED** [review]

**Review by fkolar — APPROVED** [review]

**Pull request merged**